### PR TITLE
Monitoring usage of a device

### PR DESCRIPTION
**Pull request by maciej-czajkowski**
Monitoring of usage can be seen in an added window in "Windows" drop down menu.
Test driver has been added which uses the implemented proxy.

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/proxy/DeviceUsageProxy.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package edu.kis.powp.jobs2d.drivers.proxy;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.manager.DeviceUsageManager;
+
+
+public class DeviceUsageProxy implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/proxy/DeviceUsageProxy.java]: In the GoF design patterns catalog such design is called 'decorator' not 'proxy' (proxy wraps a concrete class).

File: src/main/java/edu/kis/powp/jobs2d/command/manager/DeviceUsageManager.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Manager class for the device.
+ * Can be used by multiple drivers
+ */
+public class DeviceUsageManager {
+	private final Publisher publisher = new Publisher();
+
+	private double distanceTravelled = 0.0;
+	private double distanceDeviceUsed = 0.0;
+
+	public void notifyHeadMovePerformed(double distance) {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/manager/DeviceUsageManager.java]: 'notify' is a collateral result, the operation is more about updating usage.

**Review by kowallus — CHANGES_REQUESTED** [review]
Great, but I suggest minor renaming.

File: src/main/java/edu/kis/powp/jobs2d/command/manager/DeviceUsageManager.java (hunk 2 above)
Comment (mbarancewicz) [reply src/main/java/edu/kis/powp/jobs2d/command/manager/DeviceUsageManager.java]: renamed

File: src/main/java/edu/kis/powp/jobs2d/drivers/proxy/DeviceUsageProxy.java (hunk 1 above)
Comment (mbarancewicz) [reply src/main/java/edu/kis/powp/jobs2d/drivers/proxy/DeviceUsageProxy.java]: renamed

**Comment by kowallus** [comment]
You can claim a new task in trello.

**Pull request merged**